apps/src/javalab/JavalabEditor.jsx

```diff
@@ -23,32 +25,41 @@ class JavalabEditor extends React.Component {
     setEditorText: PropTypes.func
   };
 
-  onChange = newValue => {
-    this.props.setEditorText(newValue);
-  };
-
   componentDidMount() {
-    let textInput = $('.ace_text-input');
-    if (textInput) {
-      let textInputElement = textInput.first();
-      textInputElement.attr('aria-label', 'java editor panel');
-    }
+    this.editor = new EditorView({
+      state: EditorState.create({
+        extensions: [
+          basicSetup,
+          java(),
+          keymap.of([defaultTabBinding]),
+          oneDarkTheme,
+          oneDarkHighlightStyle
+        ]
+      }),
+      parent: this._codeMirror,
+      dispatch: this.dispatchEditorChange()
+    });
   }
 
+  dispatchEditorChange = () => {
+    return tr => {
```

Comment:
What is `tr` in this case?

Reply:
tr is a codemirror [transaction](https://codemirror.net/6/docs/ref/#state.Transaction) I'll add a comment to make it more clear

Reply:
👍